src/m365/planner/commands/plan/plan-set.ts

```diff
@@ -0,0 +1,156 @@
+import { Cli, CommandOutput, Logger } from '../../../../cli';
+import Command, {
+  CommandOption,
+  CommandError
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import Utils from '../../../../Utils';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import commands from '../../commands';
+import request from '../../../../request';
+import { Options as PlanGetCommandOptions } from '../plan/plan-get';
+import * as planGetCommand from '../plan/plan-get';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  title?: string;
+  newTitle: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerPlanSetCommand extends GraphItemsListCommand<any> {
+  public get name(): string {
+    return commands.PLAN_SET;
+  }
+
+  public get description(): string {
+    return 'Set a Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (error?: any) => void): void {
+    const getPlan = (): Promise<CommandOutput> => {
+      if (this.verbose) {
+        logger.logToStderr(`Retrieving plan ${args.options.id || args.options.title} ...`);
+      }
+
+      const options: PlanGetCommandOptions = {
+        ...args.options,
+        output: 'json',
+        debug: this.debug,
+        verbose: this.verbose
+      };
+
+      return Cli.executeCommandWithOutput(planGetCommand as Command, { options: { ...options, _: [] } });
+    };
+
+    const updatePlan = (plan: any): Promise<void> => {
+      if (this.verbose) {
+        logger.logToStderr(`Updating plan with id ${plan['id']} ...`);
+      }
+
+      const requestOptions: any = {
+        url: `${this.resource}/v1.0/planner/plans/${plan['id']}`,
+        headers: {
+          'accept': 'application/json',
+          'If-Match': `${plan["@odata.etag"]}`
+        },
+        responseType: 'json',
+        data: {
+          title: args.options.newTitle
+        }
+      };
+
+      return request.patch(requestOptions);
+    };
+
+    getPlan()
+      .then((output: CommandOutput): void => {
+        if (!output.stdout) {
+          cb(new CommandError(`No plan found`));
+          return;
+        }
+        let plans: any = JSON.parse(output.stdout);
+        if(!Array.isArray(plans)) {
+          plans = [plans];
+        }
+        const promises = [];
```

Comment:
Why are we defining an array of promises for updating one plan?

Reply:
If we have multiple plans (in the same group) with the same name, we update them all. That's why we have an array of promises.

Reply:
Since we settled on retrieving a single plan, I think we can refactor this to update just one plan, right?

Reply:
Yes, only one plan will be updated. I'm refactoring the code, so we don't have an array of promises.